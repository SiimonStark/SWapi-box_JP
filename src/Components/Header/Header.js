import React from 'react';
import Nav from '../Nav/Nav';
import PropTypes from 'prop-types';

const Header = ({updateTab}) => {
  return (
    <header>
      <h1>SWapi Box</h1>
      <Nav updateTab={updateTab} />
    </header>
  )
}

Header.propTypes = {
  updateTab: PropTypes.func.isRequired
}

export default Header;