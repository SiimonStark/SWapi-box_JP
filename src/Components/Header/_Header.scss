header {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  background: url(./images/moonDark.gif);
  background-size: 100% 500%;
  background-position: center;

  h1 {
    margin: 10px;
  }
}