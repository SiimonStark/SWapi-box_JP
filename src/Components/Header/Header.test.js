import React from 'react';
import {shallow} from 'enzyme';
import Header from './Header';

let updateTabMock = jest.fn();

describe('Header', ()=>{
  let wrapper;
  beforeEach(()=>{
    wrapper = shallow(<Header updateTab={updateTabMock}/>)
  });

  it('should match snapshot', ()=> {
    expect(wrapper).toMatchSnapshot();
  });
})