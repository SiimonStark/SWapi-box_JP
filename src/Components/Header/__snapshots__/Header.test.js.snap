// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<header>
  <h1>
    SWapi Box
  </h1>
  <Nav
    updateTab={[MockFunction]}
  />
</header>
`;
